### PR TITLE
core: Convert some functions to new code style

### PR DESCRIPTION
**Pull request by cgwalters**
Not doing the whole file since it's big, these are some simple ones to keep
momentum up.

### CODE REVIEW

File: src/libostree/ostree-core.c [hunk 1]
```diff
@@ -830,22 +790,18 @@ ostree_checksum_file_from_input (GFileInfo        *file_info,
...
           if (!ot_gio_splice_update_checksum (NULL, in, checksum, cancellable, error))
-            goto out;
+            return FALSE;
         }
     }
 
-  ret_csum = ot_csum_from_gchecksum (checksum);
-
-  ret = TRUE;
+  g_autofree guchar *ret_csum = ot_csum_from_gchecksum (checksum);
```
Comment (jlebon) [inline src/libostree/ostree-core.c]: We can just assign it directly to the out var at this point, right?

Comment (cgwalters) [inline src/libostree/ostree-core.c]: In this case, yes.  As a general rule we have to be a bit careful since some functions have nullable out arguments.

**Comment by jlebon** [comment]
@rh-atomic-bot r+ 57b6719

**Comment by alexlarsson** [comment]
Check out FlatpakRepoTransaction inhttps://github.com/flatpak/flatpak/blob/master/common/flatpak-utils.h too. Very useful autoptr thing for ostree commits.

**Pull request closed without merging**